imageCompress/crop.py

```diff
@@ -1,18 +1,69 @@
-def crop(image, height, width):
+
+import numpy as np
+import os
+from skimage.io import imread, imsave
+
+def crop(img_path, H, W):
     """
     Function to crop images
-
     Parameters:
     ----------
-    image, a 3d array of the image (obtained using plt.imread(image))
-    height, integer, the desired height of the cropped image
-    width, integer, the desired width of the cropped image
-
+    img_path - File path of the image .
+    H        - Integer, the desired height of the cropped image
+    W        - Integer, the desired width of the cropped image
+    
     Returns:
-    --------
-    cropped_image, a 3d array with dimensions height x width x 3
-
+    -------
+    cropped image path and saves the image .
+    
     Example:
-    crop(image, 10, 15) returns an array with shape (10, 15, 3)
+    -------
+    crop("data/sample.png", 10, 15) 
+    saves the image and returns the path of the cropped image.
     """
-    pass
+    #----------------------------------------Initialisation-------------------------------------------------------#
+    image  = imread(img_path)   
+    height = image.shape[0]-H
+    width  = image.shape[1]-W
+    #---------------------------------------Exception Handling----------------------------------------------------#
+    # Exception handling for input validation like Type error, invalid values , unrealistic desired dimension     #
+    #-------------------------------------------------------------------------------------------------------------#
+    myError = False
+    if type(H) != int or type(W) !=int or type(image) is not np.ndarray:
```

Comment:
If you read the image using imread, it will always be a numpy array. I guess it would be better to check if the input path is a string or not here.

Reply:
Done